### PR TITLE
ticket 7, post request

### PR DESCRIPTION
**Pull request by Nicolakirk**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -179,6 +179,32 @@ test("status 200 - returns the comments in  descending order ", () => {
...
+                comment_id: expect.any(Number),
+                votes: 0,
+                created_at: expect.any(String),
+                author: 'rogersop',
+               body: "This is what I want to write.",
+               article_id: 1,
+            })
+        })
+    })
+
```
Comment (NRMSMITH) [inline __tests__/app.test.js:205]: 201: ignores unnecessary properties
400: invalid ID
404: non-existent ID
400: missing required fields (i.e. no username or no body)
404: username does not exist

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Your code is clean, but you're missing about 5 tests. Please do the testing and then resubmit this pr :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -202,9 +203,79 @@ describe("POST /api/articles/:article_id/comments test ", ()=>{
...
+    const inputComment = {
+        username: 'rogersop',
+        body: 'This is what I want to write.', 
+        }
+        return request(app)
+        .post("/api/articles/2002/comments")
+        .send(inputComment)
+        expect(404)
+    .then(({ body }) => {
+     expect(body.message).tobe("Invalid article Id");
```
Comment (NRMSMITH) [inline __tests__/app.test.js:277]: I think 'not found' is more appropriate for the message in this case.

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Make sure you're following the tests being suggested. Remember 404 means that it's valid but can't be found, and 400 means the user has done something wrong.

File: controllers/error_controllers.js [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
         res.status( err.status).send({msg: err.msg});
 
     }else {
         next(err);
     }
 };
+exports.handlePSQL400s = (err, req, res, next) =>{
+    if (err.code === '22P02'){
+        res.status(400).send({msg:'Bad Request'})
+    }else if (err.code === '23503') {
```
Comment (popatre) [inline controllers/error_controllers.js:18]: Off the top of my head, i think this error code is for foreign key issues, so might be better as `not found`

**Review by popatre — CHANGES_REQUESTED** [review]
Hello. Lovely stuff, could you just remove the .only and change the not existing username test to send back a 404 ?

**Review by itsaliabdallah — APPROVED** [review]

**Pull request merged**